From: Balint Reczey <balint.reczey@canonical.com>
Date: Wed, 22 Jan 2020 16:12:03 +0100
Subject: Revert "network: if /sys is rw, then udev should be around"

The change broke networking in LXD containers

This reverts commit 96d7083c5499b264ecebd6a30a92e0e8fda14cd5.

LP: #1857123
---
 src/network/networkd-link.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/network/networkd-link.c b/src/network/networkd-link.c
index ee5f0f2..3ac3507 100644
--- a/src/network/networkd-link.c
+++ b/src/network/networkd-link.c
@@ -69,6 +69,7 @@
 #include "tmpfile-util.h"
 #include "tuntap.h"
 #include "udev-util.h"
+#include "virt.h"
 #include "vrf.h"
 
 bool link_ipv6_enabled(Link *link) {
@@ -1433,7 +1434,7 @@ static int link_check_initialized(Link *link) {
 
         assert(link);
 
-        if (!udev_available())
+        if (detect_container() > 0)
                 return link_initialized_and_synced(link);
 
         /* udev should be around */
